source/common/router/router.h

```diff
@@ -229,8 +229,8 @@ class FilterConfig {
 
 using FilterConfigSharedPtr = std::shared_ptr<FilterConfig>;
 
-class UpstreamRequest;
```

Comment:
I'm going to make finding a better name optional since this was already UpstreamRequest, but given we're doing all these response actions I wonder if there's a better stream-based name.